### PR TITLE
Restore "Shut down executor services after tests"

### PR DESCRIPTION
**Pull request by swankjesse**
This reverts commit 9178dcc2cf419da1f8e4903c54c093a076233657.

### CODE REVIEW

File: misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt [hunk 1]
```diff
@@ -65,38 +68,56 @@ internal class RealTransacter private constructor(
...
-              val shards = s.executeQuery("SHOW VITESS_SHARDS")
-                  .map { rs -> Shard.parse(rs.getString(1)) }
-                  .toSet()
-              if (shards.isEmpty()) {
-                throw SQLRecoverableException("Failed to load list of shards")
+  /**
+   * Uses a dedicated thread to query Vitess for the database's current shards. This caches the
+   * set of shards for 5 minutes.
+   */
+  class ShardListFetcher {
```
Comment (swankjesse) [inline misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt]: @jjestrel @tirsen FYI!

File: misk-core/src/main/kotlin/misk/concurrent/ExecutorServiceFactory.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * Inject this rather than using the [Executors] factory class to create thread pools. Executors
+ * created with this factory will automatically be shut down when the service or test completes.
+ *
+ * For all functions, `nameFormat` a string in the format specified by
+ * [ThreadFactoryBuilder.setNameFormat], like "rpc-pool-%d". If the string has a single `%d`
+ * placeholder it will be assigned sequentially. Omit the placeholder if only one thread will be
+ * required.
+ */
+interface ExecutorServiceFactory {
```
Comment (swankjesse) [inline misk-core/src/main/kotlin/misk/concurrent/ExecutorServiceFactory.kt]: New API requires a name format.

File: misk-core/src/main/kotlin/misk/concurrent/ExecutorServiceFactory.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+
+/**
+ * Inject this rather than using the [Executors] factory class to create thread pools. Executors
+ * created with this factory will automatically be shut down when the service or test completes.
+ *
+ * For all functions, `nameFormat` a string in the format specified by
+ * [ThreadFactoryBuilder.setNameFormat], like "rpc-pool-%d". If the string has a single `%d`
+ * placeholder it will be assigned sequentially. Omit the placeholder if only one thread will be
```
Comment (mpawliszyn) [inline misk-core/src/main/kotlin/misk/concurrent/ExecutorServiceFactory.kt]: Should we assert the placeholder when asking for pools other than the single thread one.

File: misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt (hunk 1 above)
Comment (jjestrel) [reply misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt]: nit: should we just have two and bind the one we need?

File: misk-core/src/main/kotlin/misk/concurrent/ExecutorServiceFactory.kt (hunk 3 above)
Comment (swankjesse) [reply misk-core/src/main/kotlin/misk/concurrent/ExecutorServiceFactory.kt]: Good idea. Coming in follow-up.

File: misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt (hunk 1 above)
Comment (swankjesse) [reply misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt]: Yeah, good idea. Coming in follow-up.

File: misk-core/src/main/kotlin/misk/concurrent/ExecutorServiceFactory.kt (hunk 3 above)
Comment (swankjesse) [reply misk-core/src/main/kotlin/misk/concurrent/ExecutorServiceFactory.kt]: https://github.com/cashapp/misk/pull/1405

**Pull request merged**